viewer/src/utilities/metrics.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+declare let VERSION: string;
+declare let MIXPANEL_TOKEN: string;
+
+import mixpanel from 'mixpanel-browser';
+
+type TrackedEvents = 'init' | 'addModel' | 'error';
+type EventProps = {
+  [key: string]: any;
+  // names mentioned instead of just `string` type for typo protection,
+  // better than nothing, easier than anything else
+  moduleName:
+    | 'RevealManager'
+    | 'Cognite3DViewer'
+    | 'CadManager'
+    | 'CachedRepository'
+    | 'Cognite3DModel'
+    | 'LocalHostRevealManager'
+    | 'sectorUtilities';
+  methodName: string;
+};
+
+let globalLogMetrics = true;
+const globalProps = {
+  VERSION: VERSION,
+  project: 'unknown'
+};
+
+export function initMetrics(logMetrics: boolean, project: string, eventProps: EventProps) {
+  // Even though mixpanel has an opt out property, the mixpanel object
+  // used by Metrics is not available here, so we have our own way of opting out.
+  globalLogMetrics = logMetrics;
+  if (!globalLogMetrics) {
+    return;
+  }
+  mixpanel.init(MIXPANEL_TOKEN, { persistence: 'localStorage' });
+  globalProps.VERSION = VERSION;
+  if (project) {
+    globalProps.project = project;
+  }
+
+  trackEvent('init', eventProps);
+}
```

Comment:
Just to make sure, does this handle when Mixpanel is blocked, e.g. by an exentsion? Just making sure that Adblock or Disconnect extensions etc doesn't cause Reveal to fail.

Reply:
@larsmoa good point. I double-checked that. And no, mixpanel isn't so poorly designed, so we don't need to cover it with try/catch every time we use it. It won't throw in case there is a network error, or even a wrong token passed.